backend/src/queue/tests/queue.e2e.spec.ts

```diff
@@ -82,21 +98,117 @@ describe("User", () => {
       });
     });
 
-    it(`unauthorized`, (done) => {
+    it(`Unauthorized`, (done) => {
       const clientSocket = io(`ws://localhost:${address.port}/queue`);
       clientSocket.connect();
       clientSocket.on("connect_error", () => {
         done();
       });
     });
 
-    afterAll(() => {
-      app.getHttpServer().close();
+    it(`Duplicate connection`, (done) => {
+      const clientSocket1 = io(`ws://localhost:${address.port}/queue`, {
+        extraHeaders: {
+          Authorization: "true",
+          user: "1",
+        },
+      });
+      const clientSocket2 = io(`ws://localhost:${address.port}/queue`, {
+        extraHeaders: {
+          Authorization: "true",
+          user: "1",
+        },
+      });
+
+      clientSocket1.connect();
+
+      // To prevent race condition.
+      setTimeout(() => clientSocket2.connect(), 100);
+
+      clientSocket1.on("exception", ({ message }) => {
+        expect(message).toBe("Duplicate connection");
+        clientSocket1.disconnect();
+        done();
+      });
+
+      clientSocket2.on("connect", () => {
+        clientSocket2.disconnect();
+      });
+    });
+  });
+
+  describe("Handle find", () => {
```

Comment:
Could we add a test to make sure that the `createRoom` is called with the correct arguments?

Reply:
Done